src/modules/plus/filter_pillar_echo.c

```diff
@@ -414,7 +251,11 @@ static int filter_get_image( mlt_frame frame, uint8_t **image, mlt_image_format
 	uint8_t* dst = mlt_pool_alloc( size );
 
 	bilinear_scale_rgba( *image, dst, *width, *height, rect );
-	box_blur( dst, dst, *width, *height, blur );
+	if (blur != 0) {
+		struct mlt_image_s img;
+		mlt_image_set_values( &img, dst, *format, *width, *height );
+		mlt_image_box_blur( &img, blur, blur );
```

Comment:
To avoid code duplication the pillar_echo filter should encapsulate a blur filter that can be changed from its default via a string property (e.g. "blur_filter") including properties that begin with "blur." for example to set properties on the nested filter. This is a pattern we have used elsewhere.

Reply:
I am familiar with the service encapsulation method. I do not like it for this case because of the extra resource overhead and complication:
* Requires the construction of an entire extra service and passing parameters using the properties
* Requires constructing an extra frame to pass the image into the service
* Does not support frame threading unless the parameters are attached to the frame and interpreted by the encapsulated filter
All this extra complication and overhead - and all I want to do is have a single function that is shared by two files.

I will move this function into a shared file at the module level as we have done for other modules:
https://github.com/mltframework/mlt/tree/master/src/modules/plus/ebur128
https://github.com/mltframework/mlt/blob/master/src/modules/plusgpl/utils.h
https://github.com/mltframework/mlt/blob/master/src/modules/qt/graph.h